### DISCUSSION TITLE
Esp32 flavour ⚒️

### DISCUSSION THREAD

**Discussion by jvegaf** [issalig/toshiba_air_cond#25] [Show and tell]
Build finished 

![PXL_20220915_145615290.jpg](https://user-images.githubusercontent.com/1075386/193456681-bc7512f5-2c64-4f67-bd09-dca6fa06ef76.jpg)

![PXL_20220915_154305295.MP.jpg](https://user-images.githubusercontent.com/1075386/193456683-9f38f66f-6d04-4b0f-af9b-6c70e30b3851.jpg)

![PXL_20221002_133052842.MP.jpg](https://user-images.githubusercontent.com/1075386/193456740-41ec8784-eac7-4ef6-9f73-677c1a3a425e.jpg)

**Comment by issalig** [comment]
It looks great!
Do you need to power it from 5v or just AB wires?

**Reply by jvegaf** [reply]
in this version only can power it with USB connection or with 5v with the power connector ☹️ .  I'll try get the power from AB wires using extra circuit board, pretty similar to @burrocargado solution.

On the next design rev. will have that feature.

Also I'm refactoring the esp32 firmware code, splitting and extracting the code to modules, trying to do more testable and robust code. This will give the option in the future, to have great things like CI

**Comment by dzungpv** [comment]
@jvegaf what about your progress? I am interest with power take direct from the AB bus.
